app.js

```diff
@@ -276,6 +276,10 @@ models.sequelize.authenticate().then(function () {
     } else {
       throw new Error('server still not ready after db synced')
     }
+    // Allow to timeout-lock guest notes so they can no longer be changed after the timeout period
+    if (config.lockGuestNotes.timeout > 0) {
+      setInterval(models.Note.lockOldGuestNotes, (config.lockGuestNotes.timeout / 2) * 1000, config.lockGuestNotes.timeout, config.lockGuestNotes.permission)
```

Comment:
worst case, this is off by half of `timeout / 2`. So I can't actually be certain that 1 day is actually one day, it might be just under 36 hours. Depending on how important this is, we could query the newest guest note and set a `setTimeout` specifically for its timout time. We would also only have to perform this check once at server start and once per created note.
Again: this might be overkill.

Reply:
I would rather go for a time constant than for an individual timeout, since it should be less stressful for the node application.

Unless someone explicitly complaints about the timeout inaccuracy, it should be fine. The original idea of the implementation was based on using a cron-job, so I would say we are close enough 😉

Reply:
that would be fixed by running it once on startup.

Reply:
But if I don't restart the app that often this also doesn't help. Why don't you check the database every e.g. 60 minutes?